sympy/polys/constructor.py

```diff
@@ -122,8 +123,12 @@ def _construct_composite(coeffs, opt):
         return None
 
     if opt.composite is None:
-        if any(gen.is_number for gen in gens):
-            return None # generators are number-like so lets better use EX
+        for gen in gens:
+            if gen.is_number:
+                if isinstance(gen, NumberSymbol) and not gen.is_algebraic:
+                    continue
+                # generators are number-like so lets better use EX
```

Comment:
It is probably not necessary to test for NumberSymbols explicitly. That could possibly give
```
    if any(gen.is_number and gen.is_algebraic for gen in gens):
        return None
```